### ISSUE TITLE
[BUG] context.close() hangs when using together with signal

### ISSUE THREAD

**Issue by chenstrace** [microsoft/playwright-python#1843]
### System info
- Playwright Version: [v1.27.0]
- Operating System: [All]
- Browser: [All, Chromium, Firefox, WebKit]
- Other info:

### Source code

```python
import signal
import time

from playwright.sync_api import sync_playwright

def my_sig_handler(signum, frame):
    global exit
    exit = True


exit = False
playwright = sync_playwright().start()
browser = playwright.chromium.launch(headless=False)
context = browser.new_context(record_har_path="/Users/chenstrace/example.har")
page = context.new_page()
page.goto("https://www.baidu.com/")
signal.signal(signal.SIGINT, my_sig_handler)

try:
    while True:
        if exit:
            break
        else:
            print("sleep 1", flush=True)
            time.sleep(1)
finally:
    print("close context", flush=True)
    context.close()
    print("close browser", flush=True)
    browser.close()
```

**Steps**
- save the code to file a.py
- run the program in terminal, using  ```python3 a.py```
- wait for 5 seconds
- press CTRL+C in terminal.

**Expected**
- python program can exit
- a har file was saved to /Users/chenstrace/example.har

**Actual**
- the program hangs on context.close()
- no har file was saved.

**Comment by mxschmitt** [comment]
I can reproduce.

**Comment by mxschmitt** [comment]
The issue is that if you press Command + C, the driver gets killed and then https://github.com/microsoft/playwright-python/blob/b913f4830f065883419e6840e3f5be089c47cb08/playwright/_impl/_browser_context.py#L448 will stall since we don't cancel the callbacks when the driver connection gets closed.

**Comment by mxschmitt** [comment]
This will be fixed with 1.34, thanks for filing the bug!

**Comment by B-2U** [comment]
hello there, I just tested the same code in [playwright==1.34.0]
but its still happened
the program hangs on context.close()
